pycode/memilio-epidata/memilio/epidata/geoModificationGermany.py

```diff
@@ -560,7 +561,8 @@ def merge_df_counties(
         merged_id rows.
     """
     # ensure that separated_ids and dataframe ids can be compared
-    if type(separated_ids[0]) != type(df[dd.EngEng['idCounty']][0]):
+    # df column should be an int, as seperated_ids and merged_id are int.
+    if not isinstance(df[dd.EngEng['idCounty']][0], (int, np.integer)):
```

Comment:
Wouldn't it be easier to use df.dtypes e.g. df[dd.EngEng['idCounty']].dtypes['idCounty'] == 'int64'?

Reply:
the problem is, that depending on how the dataframe is created, the idCounty column can have int32, int64 or np.int, np.int_
This should catch all and only change the dtype if the idCounty column is object or string

Reply:
okay